packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -636,10 +646,47 @@ class S3Source extends Source {
   }
 }
 
+/**
+ * Common properties between {@link GitHubSource} and {@link GitHubEnterpriseSource}.
+ */
+interface CommonGithubSourceProps extends ThirdPartyGitSourceProps {
+  /**
+   * This parameter is used for the `context` parameter in the GitHub commit status.
+   *
+   * @example "My build #$CODEBUILD_BUILD_NUMBER"
+   * @default "AWS CodeBuild $AWS_REGION ($PROJECT_NAME)"
+   */
```

Comment:
Let's also add links to the CodeBuild docs about this functionality:

```suggestion
   *
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.context
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
   */
```